tests/jax/components/training/step_test.py

```diff
@@ -0,0 +1,723 @@
+        rewards[:-1], discounts[:-1], 0.95, values
+    )
+    advantages = jax.lax.stop_gradient(advantages)
+
+    # Exclude the bootstrap value
+    target_values = values[:-1] + advantages
+    target_values = jax.lax.stop_gradient(target_values)
+
+    return advantages, target_values
+
+
+def epoch_update(carry: Tuple, unused_t: Tuple[()]) -> Tuple:
+    """Performs model updates based on one epoch of data."""
+    return carry, {}
+
+
+class MockTrainerLogger:
+    """Mock of TrainerLogger to test DefaultTrainerStep component"""
+
+    def __init__(self) -> None:
+        self.written = None
+
+    def write(self, results: Any) -> None:
+        self.written = results
+
+
+class MockParameterClient:
+    """Mock of ParameterClient to test DefaultTrainerStep component"""
+
+    def __init__(self) -> None:
+        self.params = {
+            "trainer_steps": 0,
+            "trainer_walltime": -1,
+        }
+        self.call_set_and_get_async = False
+
+    def add_async(self, params: Any) -> None:
+        self.params = params
+
+    def set_and_get_async(self) -> None:
+        self.call_set_and_get_async = True
+
+
+class MockTrainer(Trainer):
+    """Mock of Trainer"""
+
+    def __init__(self) -> None:
+        trainer_agent_net_keys = {
+            "agent_0": "network_agent_0",
+            "agent_1": "network_agent_1",
+            "agent_2": "network_agent_2",
+        }
+        networks = {
+            "networks": {
+                "network_agent_0": SimpleNamespace(
+                    params={"key": jnp.array([0.0, 0.0, 0.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+                "network_agent_1": SimpleNamespace(
+                    params={"key": jnp.array([1.0, 1.0, 1.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+                "network_agent_2": SimpleNamespace(
+                    params={"key": jnp.array([2.0, 2.0, 2.0])},
+                    network=SimpleNamespace(apply=apply),
+                ),
+            }
+        }
+        opt_states = {
+            "network_agent_0": 0,
+            "network_agent_1": 1,
+            "network_agent_2": 2,
+        }
+        store = SimpleNamespace(
+            dataset_iterator=iter([1, 2, 3]),
+            step_fn=step_fn,
+            timestamp=1657703548.5225394,  # time.time() format
+            trainer_parameter_client=MockParameterClient(),
+            trainer_counts={"next_sample": 2},
+            trainer_logger=MockTrainerLogger(),
+            sample_batch_size=2,
+            sequence_length=3,
+            trainer_agent_net_keys=trainer_agent_net_keys,
+            networks=networks,
+            gae_fn=gae_advantages,
+            opt_states=opt_states,
+            key=jax.random.PRNGKey(5),
+            num_minibatches=1,
+            epoch_update_fn=epoch_update,
+            num_epochs=2,
+        )
+        self.store = store
+
+
+class DummySample:
```

Comment:
Thank you for all these updates! Just because the dummy sample class is now so big can we maybe move this to a separate file in the same directory and import it from there to make the test a bit cleaner to read?